src/accelerate/data_loader.py

```diff
@@ -19,6 +19,12 @@
 import torch
 from torch.utils.data import BatchSampler, DataLoader, IterableDataset, RandomSampler
 
+from accelerate.utils.imports import is_torchdata_stateful_dataloader_available
+
+
+if is_torchdata_stateful_dataloader_available():
+    from torchdata.stateful_dataloader import StatefulDataLoader
```

Comment:
To keep base imports fast, checks like this should happen directly when needed. In this case the import should occur during`DataLoaderAdapter`'s `__init__`

Reply:
Done, but I think I also need to put this import in `skip_first_batches` since there's a usage of StatefulDataLoader there as well.